common/grpc/src/dns_resolver.rs

```diff
@@ -153,7 +153,8 @@ impl ConnectionFactory {
         inner_connector.set_keepalive(None);
         inner_connector.enforce_http(false);
 
```

Comment:
add `innner.set_connect_timeout` may be more suitable

Reply:
timeout is set here https://github.com/datafuselabs/databend/pull/5616/files#diff-50b6563436352a0bf2e50ea49d02a3883b8a9478637144963bae177bd359d49cL197